src/custom-element.js

```diff
@@ -591,13 +589,158 @@ function createBaseCustomElementClass(win, elementConnectedCallback) {
+              ? ReadyState.LOADING
+              : ReadyState.MOUNTING
+          );
+          return this.impl_.mountCallback(signal);
+        })
+        .then(() => {
+          this.mountAbortController_ = null;
+          if (signal.aborted) {
+            throw cancellation();
+          }
+          this.signals_.signal(CommonSignals.MOUNTED);
+          if (!this.implClass_.usesLoading(this)) {
+            this.setReadyStateInternal(ReadyState.COMPLETE);
+          }
+        })
+        .catch((reason) => {
+          this.mountAbortController_ = null;
+          if (isCancellation(reason)) {
+            this.mountPromise_ = null;
+          } else {
+            this.signals_.rejectSignal(
+              CommonSignals.MOUNTED,
+              /** @type {!Error} */ (reason)
+            );
+            this.setReadyStateInternal(ReadyState.ERROR, reason);
+          }
+          throw reason;
+        }));
+    }
+
+    /**
+     * Requests the element to be mounted as soon as possible.
+     * @return {!Promise}
+     * @final
+     */
+    mount() {
+      if (this.mountPromise_) {
+        return this.mountPromise_;
+      }
+
+      // Create the abort controller right away to ensure that we the unmount
+      // will properly cancel this operation.
+      this.mountAbortController_ =
+        this.mountAbortController_ || new AbortController();
+      const {signal} = this.mountAbortController_;
+
+      const readyPromise = this.signals_.whenSignal(
+        CommonSignals.READY_TO_UPGRADE
+      );
+      return readyPromise.then(() => {
+        if (!this.V1()) {
+          return this.whenBuilt();
+        }
+        if (signal.aborted) {
+          throw cancellation();
+        }
+        if (this.mountPromise_) {
+          return this.mountPromise_;
+        }
+        const scheduler = getSchedulerForDoc(this.getAmpDoc());
+        scheduler.scheduleAsap(this);
+        return this.whenMounted();
+      });
+    }
+
+    /**
+     * Unmounts the element and makes it ready for the next mounting
+     * operation.
+     * @final
+     */
+    unmount() {
+      // Ensure that the element is paused.
+      if (this.isConnected_) {
+        this.pause();
+      }
+
+      // Legacy pre-V1 elements simply unload.
+      if (!this.V1()) {
+        this.getResource_().unlayout();
+        return;
+      }
+
+      // Cancel the currently mounting operation.
+      if (this.mountAbortController_) {
+        this.mountAbortController_.abort();
+        this.mountAbortController_ = null;
+      }
+
+      // Unschedule a currently pending mount request.
+      const scheduler = getSchedulerForDoc(this.getAmpDoc());
+      scheduler.unschedule(this);
+
+      // Try to unmount. Not every element has been mounted, can be unmounted
+      // or has anything to unmount.
+      const reMount =
+        !this.mountPromise_ || !this.impl_ || this.impl_.unmountCallback();
+      if (!reMount) {
+        return;
+      }
```

Comment:
Shouldn't this be "if mounted, and impl, and `mountCallback` returns true"?  Right now (if I expand out the de Morgan's law): "if mounted, and impl, and `unmountCallback` returns **false**".

```suggestion
      const reMount =
        this.mountPromise_ && this.impl_ && this.impl_.unmountCallback();
      if (!reMount) {
        return;
      }
```

Reply:
The initial meaning was that the mounting should be scheduled in all cases, except when `this.unmountCallback() === false`. That being said, this is obviously confusing. I decided that we will have more problems than it's worth from inheriting this feature from the `unlayoutCallback`. So in the latest commit I've dropped it. I think now it looks much clearer. PTAL.